src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.GenerateAssemblyInfo.targets

```diff
@@ -0,0 +1,120 @@
+    <!-- Determines if the generated Razor assembly includes an auto-generated assembly info. -->
+    <GenerateRazorTargetAssemblyInfo Condition="'$(GenerateRazorTargetAssemblyInfo)'==''">$(GenerateAssemblyInfo)</GenerateRazorTargetAssemblyInfo>
+    <GenerateRazorTargetAssemblyInfo Condition="'$(GenerateRazorTargetAssemblyInfo)'==''">true</GenerateRazorTargetAssemblyInfo>
+
+    <!-- Set to true, to automatically include some AssemblyAttributes inferred from the project metadata in the generated Razor assembly -->
+    <EnableDefaultRazorTargetAssemblyInfoAttributes Condition="'$(EnableDefaultRazorTargetAssemblyInfoAttributes)'==''">true</EnableDefaultRazorTargetAssemblyInfoAttributes>
+
+    <GeneratedRazorTargetAssemblyInfo Condition="'$(GeneratedRazorTargetAssemblyInfo)'==''">$(IntermediateOutputPath)$(MSBuildProjectName).RazorAssemblyInfo.cs</GeneratedRazorTargetAssemblyInfo>
+
+    <_GeneratedRazorAssemblyInfoInputsCacheFile>$(IntermediateOutputPath)$(MSBuildProjectName).RazorAssemblyInfoInputs.cache</_GeneratedRazorAssemblyInfoInputsCacheFile>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <GenerateRazorAssemblyInfoDependsOn>
+      GetRazorAssemblyAttributes;
+      CreateRazorGeneratedAssemblyInfoInputsCacheFile;
+    </GenerateRazorAssemblyInfoDependsOn>
+  </PropertyGroup>
+
+  <Target 
+    Name="GenerateRazorAssemblyInfo" 
+    DependsOnTargets="$(GenerateRazorAssemblyInfoDependsOn)"
+    Inputs="$(_GeneratedRazorAssemblyInfoInputsCacheFile)"
+    Outputs="$(GeneratedRazorTargetAssemblyInfo)"
+    Condition="'$(GenerateRazorTargetAssemblyInfo)'=='true' AND '@(RazorAssemblyAttribute)'!=''">
+
+    <ItemGroup>
+      <!-- Ensure the generated assemblyinfo file is not already part of RazorCompile sources -->
+      <RazorCompile Remove="$(GeneratedRazorTargetAssemblyInfo)" />
+      <RazorCompile Include="$(GeneratedRazorTargetAssemblyInfo)" />
+    </ItemGroup>
+
+    <WriteCodeFragment AssemblyAttributes="@(RazorAssemblyAttribute)" Language="C#" OutputFile="$(GeneratedRazorTargetAssemblyInfo)" />
+
+    <ItemGroup>
+      <FileWrites Include="$(GeneratedRazorTargetAssemblyInfo)" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target Name="GetRazorAssemblyAttributes" DependsOnTargets="GetAssemblyVersion">
+
+    <PropertyGroup>
+      <RazorAssemblyFileVersion Condition="'$(RazorAssemblyFileVersion)' == ''">$(FileVersion)</RazorAssemblyFileVersion>
+      <RazorAssemblyInformationalVersion Condition="'$(RazorAssemblyInformationalVersion)' == ''">$(InformationalVersion)</RazorAssemblyInformationalVersion>
+      <RazorAssemblyTitle Condition="'$(RazorAssemblyTitle)'==''">$(RazorTargetName)</RazorAssemblyTitle>
+      <RazorAssemblyVersion Condition="'$(RazorAssemblyVersion)' == ''">$(AssemblyVersion)</RazorAssemblyVersion>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <RazorAssemblyAttribute Include="System.Reflection.AssemblyCompanyAttribute" Condition="'$(Company)' != '' and '$(GenerateAssemblyCompanyAttribute)' == 'true'">
```

Comment:
We need a way to generate an attribute. Formalizing this helps us get rid of some of the code in https://github.com/aspnet/Identity/blob/dev/src/UI/Microsoft.AspNetCore.Identity.UI.targets#L11.